services/management/ia/rules_service.go

```diff
@@ -355,6 +352,87 @@ func (s *RulesService) CreateAlertRule(ctx context.Context, req *iav1beta1.Creat
 	return &iav1beta1.CreateAlertRuleResponse{RuleId: rule.ID}, nil
 }
 
+func (s *RulesService) processRuleParameters(param []*iav1beta1.RuleParam, templateName string) (models.RuleParams, error) {
+	ruleParams, err := convertRuleParamsToModel(param)
+	if err != nil {
+		return nil, err
+	}
+
+	t, ok := s.templates.getTemplates()[templateName]
+	if !ok {
+		return nil, status.Errorf(codes.NotFound, "Unknown template %s.", templateName)
+	}
+
+	unknownParams := make(map[string]bool)
+
+	res := make(models.RuleParams, 0, len(t.Params))
+	for _, tp := range t.Params {
+		var filled bool
+		for _, rp := range ruleParams {
+			if _, ok := unknownParams[rp.Name]; !ok {
+				unknownParams[rp.Name] = true
+			}
+
+			if rp.Name == tp.Name {
+				if string(tp.Type) != string(rp.Type) {
+					return nil, status.Errorf(codes.InvalidArgument, "Parameter %s has type %s instead of %s.", tp.Name, rp.Type, tp.Type)
+				}
+				unknownParams[rp.Name] = false
+				filled = true
+				res = append(res, rp)
+				break
+			}
+		}
+
+		if !filled {
+			if tp.Value == nil {
```

Comment:
shouldn't this be `tp.DefaultValue`?

Reply:
`tp.Value` is the default value in fact, but that naming already spread across multiple repos

Reply:
ok